File: src/api/diagnostics.ts
import { invokeCommand } from '@/api/base'

/** 崩溃日志条目（对应后端 CrashLogEntry） */
export interface CrashLogEntry {
  filename: string
  path: string
  size: number
  modified: string
}

/** 获取崩溃日志列表 */
export function listCrashLogs(): Promise<CrashLogEntry[]> {
  return invokeCommand('list_crash_logs', undefined, { source: 'SYSTEM', silent: true })
}

/** 读取指定崩溃日志内容 */
export function readCrashLog(filename: string): Promise<string> {
  return invokeCommand('read_crash_log', { filename }, { source: 'SYSTEM' })
}

/** 清除所有崩溃日志，返回清除数量 */
export function clearCrashLogs(): Promise<number> {
  return invokeCommand('clear_crash_logs', undefined, { source: 'SYSTEM' })
}

/** 写入前端错误日志（追加到 error_YYYYMMDD.log） */
export function writeErrorLog(content: string): Promise<void> {
  return invokeCommand('write_error_log', { content }, { source: 'SYSTEM', silent: true })
}

File: src-tauri/src/commands/diagnostics.rs
use crate::services::crash_reporter;

/// 获取崩溃日志列表
#[tauri::command]
pub fn list_crash_logs() -> Vec<crash_reporter::CrashLogEntry> {
    crash_reporter::list_crash_logs()
}

/// 读取指定崩溃日志内容
#[tauri::command]
pub fn read_crash_log(filename: String) -> Result<String, String> {
    crash_reporter::read_crash_log(&filename)
        .ok_or_else(|| "日志文件不存在或无法读取".to_string())
}

/// 清除所有崩溃日志
#[tauri::command]
pub fn clear_crash_logs() -> usize {
    crash_reporter::clear_crash_logs()
}

/// 写入前端错误日志（追加模式，写入 AppData/devforge/logs/error_YYYYMMDD.log）
#[tauri::command]
pub fn write_error_log(content: String) -> Result<(), String> {
    crash_reporter::append_error_log(&content)
        .map_err(|e| format!("写入错误日志失败: {}", e))
}
